File: WorldFeed/Server/src/WorldFeed.Logic/Wrapper/JsonConvertWrapper.cs
namespace WorldFeed.Logic.Wrapper
{
    using Newtonsoft.Json;

    using WorldFeed.Common.Wrappers;

    // Wrapper classes "cannot" be tested
    public class JsonConvertWrapper : IJsonConvertWrapper
    {
        public string SerializeObject(object o)
        {
            return JsonConvert.SerializeObject(o);
        }

        public string SerializeObject(object o, JsonConverter[] converters)
        {
            return JsonConvert.SerializeObject(o, converters);
        }

        public T DeserializeObject<T>(string json)
        {
            return JsonConvert.DeserializeObject<T>(json);
        }

        public T DeserializeObject<T>(string json, JsonConverter[] converters)
        {   
            var type = typeof(T);

           // json = "{\"screen_name\":\"Volen99\",\"protected\":1,\"language\":26,\"always_use_https\":true,\"discoverable_by_email\":true,\"discoverable_by_mobile_phone\":false,\"geo_enabled\":false,\"use_cookie_personalization\":true,\"allow_contributor_request\":2,\"allow_dms_from\":0,\"allow_dm_groups_from\":0,\"time_zone\":{\"name\":\"UTC\",\"tzinfo_name\":\"Etc/UTC\",\"utc_offset\":0},\"display_sensitive_media\":false,\"smart_mute\":false,\"sleep_time\":{\"enabled\":false,\"start_time\":0,\"end_time\":0},\"SleepTimeEnabled\":false,\"SleepTimeStartHour\":0,\"SleepTimeEndHour\":0,\"translator_type\":null,\"trend_location\":[{\"woeid\":2357024,\"name\":\"Atlanta\",\"country\":\"Bulgaria\",\"countryCode\":\"bg\",\"url\":\"http://where.yahooapis.com/v1/place/2357024\",\"parentid\":23424977,\"PlaceType\":7,\"placeType\":{\"name\":null,\"code\":7}}]}";

            return JsonConvert.DeserializeObject<T>(json, converters);
        }
    }
}

File: Sharebook/Client/src/app/logic/Wrapper/JsonConvertWrapper.ts
import {IJsonConvertWrapper} from "../../core/Core/Wrappers/IJConvertWrapper";
import {Injectable} from "@angular/core";

// Wrapper classes "cannot" be tested

@Injectable({
  providedIn: 'root',
})
export class JsonConvertWrapper implements IJsonConvertWrapper {
  DeserializeObject<T>(json: string): T;
  DeserializeObject<T>(json: string, converters: any[]): T;
  DeserializeObject(json: string, converters?: any[]): any /*T*/ {
    return undefined;
  }

  SerializeObject(o: object): string;
  SerializeObject(o: object, converters: any[]): string;
  SerializeObject(o: object, converters?: any[]): string {
    return "";
  }

}

// export class JsonConvertWrapper implements IJsonConvertWrapper
// {
//   private _jsonConvert: JsonConvert = new JsonConvert();
//
//     public  SerializeObject(o: object): string
//     {
//         return JsonConvert.SerializeObject(o);
//     }
//
//     public  SerializeObject(o: object, converters: JsonConverter[]): string
//     {
//         return JsonConvert.SerializeObject(o, converters);
//     }
//
//     public  DeserializeObject<T>(json: string): T
//     {
//         return this._jsonConvert.deserializeObject(json, T);
//     }
//
//     public  DeserializeObject<T>(json: string, converters: JsonConverter[]): T
//     {
//         return JsonConvert.DeserializeObject<T>(json, converters);
//     }
// }
